src/StreamJsonRpc/ActivityTracingStrategy.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Diagnostics;
+    using Microsoft;
+    using StreamJsonRpc.Protocol;
+
+    /// <summary>
+    /// Synchronizes activities as set by the <see cref="Activity"/> class over RPC.
+    /// </summary>
+    /// <seealso cref="CorrelationManagerTracingStrategy"/>
+    public class ActivityTracingStrategy : IActivityTracingStrategy
+    {
+        /// <inheritdoc/>
+        public void ApplyOutboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (Activity.Current?.IdFormat == ActivityIdFormat.W3C)
+            {
+                request.TraceParent = Activity.Current.Id;
+                request.TraceState = Activity.Current.TraceStateString;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IDisposable? ApplyInboundActivity(JsonRpcRequest request)
+        {
+            Requires.NotNull(request, nameof(request));
+
+            if (request.TraceParent is object)
+            {
+                var state = new State(new Activity(request.Method));
+                state.NewActivity.TraceStateString = request.TraceState;
+                state.NewActivity.SetParentId(request.TraceParent);
+                state.NewActivity.Start();
+                return state;
+            }
+
+            return null;
+        }
+
+        private class State : IDisposable
+        {
+            internal State(Activity newActivity)
+            {
+                this.PriorActivity = Activity.Current;
+                this.NewActivity = newActivity;
+            }
+
+            internal Activity PriorActivity { get; }
```

Comment:
I use `readonly` fields everywhere. But we do have the analyzer rules enabled that want to see fields be `private`, and these can't be.
But FWIW, I've seen the .NET team folks many times deny that auto-properties are any slower than fields for any application. The JIT will inline cheap methods, including this property accessor, so it will actually result in the same assembly code either way. Check [this out](https://sharplab.io/#v2:C4LghgzgtgNAJiA1AHwAICYCMBYAUKgZgAIMiBhIgbzyNpIBYiAxAexYAoBKKmuvgNzAAnIhGBhgAUyIBeIgDtJAdyIBlcVPaYYmTgG5efWgEt5wIgDNZojZIB0F45IA2cA7iN1T5gA7WxEvYACkIsPpJCwACe7nwAvoZEiT5CxoJSJOhqtjwenkTeEfJgztmB7N6WTq4wBWZEKWER0dzUefl0wAAWxhAO1XDWji5uiR3dvXYhTZFR1o3hs7H5Ce1GY15mRSV15sOuy56FQsWlldOL0VQA5pLAeqvxeHFAA=). Try commenting out the property retrieval then try commenting out the field retrieval. Both remove exactly 1 assembly instruction. So they're exactly equivalent. That's in Debug build. In release builds, it all disappears because we don't use the values.